src/Application.cpp

```diff
@@ -2509,3 +2511,78 @@ void Application::toggleFastForwarding(unsigned extra)
       break;
   }
 }
+
+parsedArgs Application::parseArgs(int argc, char* argv[])
+{
+  parsedArgs result;
+
+  char argCategory = '\0';  // remember the argument category, e.g. "c" for core, "s" for system, "g" for game
+  for (int i = 0; i < argc; i++) {
+    char *currentArg = argv[i];
+
+    if (strlen(currentArg) == 0) {
+      continue;
+    }
+
+    if (currentArg[0] == '-') {
+      if (strcmp(currentArg, "-c") == 0 || strcmp(currentArg, "--core") == 0) {
+        argCategory = 'c';
+      } else if (strcmp(currentArg, "-s") == 0 || strcmp(currentArg, "--system") == 0) {
+        argCategory = 's';
+      }
+      else if (strcmp(currentArg, "-g") == 0 || strcmp(currentArg, "--game") == 0) {
+        argCategory = 'g';
+      } else {
+        argCategory = '\0';
+      }
+
+      continue;
+    }
+
+    switch (argCategory)
+    {
+      case 'c':
+        result.core.assign(currentArg);
+        break;
+      case 's':
+        try {
+          result.system = std::stoi(currentArg);
+        } catch (const std::exception& e) {
+          _logger.error(TAG "error while parsing 'system' command line argument '%s' (integer expected)", currentArg);
+          result.system = 0;
+        }
+        break;
+      case 'g':
+        result.game.assign(currentArg);
+        break;
+      
+      default:
+        break;
+    }
+  }
+  return result;
+}
+
+bool Application::tryLoadContent(parsedArgs parsedArgs)
+{
+  if (parsedArgs.core.length() == 0 || parsedArgs.system == 0 || parsedArgs.game.length() == 0) {
+    return true;  // nothing to do
+  }
+
+  if (!util::exists(parsedArgs.game)) {
+    _logger.error("file not found provided in 'game' command line argument '%s'", parsedArgs.game.c_str());
+    return false;
+  }
+  
+  RA_AttemptLogin(true); // attempt login with blocking, so that we go on exactly after the login was successful
```

Comment:
`app.init()` will have already kicked off an asynchronous login by this point. You might want to reorder things so you can avoid the asynchronous login in this code path.